include/cadmium/modeling/dynamic_atomic.hpp

```diff
@@ -50,41 +51,43 @@ namespace cadmium {
          * @tparam TIME a valid TIME class to use along with the atomic model class as ATOMIC<TIME>
          */
         template<template<typename T> class ATOMIC, typename TIME>
-        class dynamic_atomic : public atomic, public ATOMIC<TIME> {
+        class dynamic_atomic : public model, public ATOMIC<TIME> {
         public:
+            using model_type=ATOMIC<TIME>;
 
-            using output_ports = typename ATOMIC<TIME>::output_ports;
-            using input_ports = typename ATOMIC<TIME>::input_ports;
+            using output_ports = typename model_type::output_ports;
+            using input_ports = typename model_type::input_ports;
 
             // Model input and output types
             using output_bags = typename make_message_bags<output_ports>::type;
             using input_bags = typename make_message_bags<input_ports>::type;
 
             dynamic_atomic() {
                 static_assert(cadmium::concept::is_atomic<ATOMIC>::value, "This is not an atomic model");
+                cadmium::concept::atomic_model_assert<ATOMIC>();
             };
 
             void external_transition(TIME e, cadmium::dynamic_message_bags dynamic_bags) {
                 // Translate from dynamic_message_bag to template dependent input_bags type.
                 input_bags bags;
                 cadmium::modeling::fill_bags_from_map(dynamic_bags, bags);
 
-                // Forwards the translated value to the wrapped ATOMIC<TIME> class method.
-                external_transition(e, bags);
+                // Forwards the translated value to the wrapped model_type class method.
+                model_type::external_transition(e, bags);
             };
 
             void confluence_transition(TIME e, cadmium::dynamic_message_bags dynamic_bags) {
                 // Translate from dynamic_message_bag to template dependent input_bags type.
                 input_bags bags;
                 cadmium::modeling::fill_bags_from_map(dynamic_bags, bags);
 
-                // Forwards the translated value to the wrapped ATOMIC<TIME> class method.
-                confluence_transition(e, bags);
+                // Forwards the translated value to the wrapped model_type class method.
+                model_type::confluence_transition(e, bags);
             };
 
-            cadmium::dynamic_message_bags output() const {
```

Comment:
Why removing the const? I don't see you changing *this below

Reply:
The problem is that I've thought you expected the output() method to be const, so in the dynamic_atomic I made this method const, but when a did the test with the accumulator model, it fails because the accumulator does not declare this method as const, so I changed this method to a non const method. I think it should be const and the accumulator should declare this method const too.

Reply:
I added the const declaration to the accumulator and filter_first_output basic models and restored the const declaration from the dynamic_atomic() method.